src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.ConstrainedExecution;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Security;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class TransportWithMessageCredentialNetTCPTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public TransportWithMessageCredentialNetTCPTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Theory]
+        [InlineData(false, "testuser@corewcf")]
+        [InlineData(true, "randomuser@corewcf")]
+        private void BasicUserNameAuth(bool isError, string userName)
+        {
+            string testString = new string('a', 3000);
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<StartUpPermissionBaseForTC>(_output).Build();
+            using (host)
+            {
+
+                host.Start();
+                System.ServiceModel.NetTcpBinding binding = ClientHelper.GetBufferedModeBinding(System.ServiceModel.SecurityMode.TransportWithMessageCredential);
+                binding.Security.Message.ClientCredentialType = System.ServiceModel.MessageCredentialType.UserName;
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(binding,
+                    new System.ServiceModel.EndpointAddress(host.GetNetTcpAddressInUse() + Startup.WindowsAuthRelativePath));
+                System.ServiceModel.Description.ClientCredentials clientCredentials = (System.ServiceModel.Description.ClientCredentials)factory.Endpoint.EndpointBehaviors[typeof(System.ServiceModel.Description.ClientCredentials)];
+                factory.Credentials.ServiceCertificate.SslCertificateAuthentication = new System.ServiceModel.Security.X509ServiceCertificateAuthentication
+                {
+                    CertificateValidationMode = System.ServiceModel.Security.X509CertificateValidationMode.None
+                };
+                clientCredentials.UserName.UserName = userName;
+                clientCredentials.UserName.Password = RandomString(10);
+                var channel = factory.CreateChannel();
+                try
+                {
+                    if (isError)
+                    {
+                        //In Framework coming as Protocol exception while in core CommunicationException.
+                        Assert.NotNull(Record.Exception(() =>
+                        {
```

Comment:
I was trying to work out why you did this. I checked documentation and it looks like `Assert.Throws<>` expects the exact exception type. `Assert.ThrowsAny<>` expects the exception type or a derived type. Record.Exception doesn't state an option on if an exception should be thrown, it's purpose is to capture an exception if it is thrown. The failure message from Assert.NotNull won't match with the intention. You should be able to use `Assert.ThrowsAny<CommunicationException>` as ProtocolException derives from CommunicationException.